src/hotspot/share/memory/virtualspace.hpp

```diff
@@ -52,6 +54,12 @@ class ReservedSpace {
                   char* requested_address,
                   bool executable);
 
+
+  char* attempt_reserve_memory_at(char* base, size_t size);
+  char* reserve_memory(size_t size);
+  char* reserve_memory_aligned(size_t size, size_t alignment);
+  void  release_memory(char* base, size_t size);
```

Comment:
I liked the old names (..._map_or_reserve_....) better. Can you please rename them back? Their whole point is multiplexing between anonymous and mmaped reservation calls. Also, in their current form they read identical to the os::... functions which is really confusing.
For release I propose `release_mapped_or_reserved_memory'. Its a mouthful but it clearly states what it does.